python/mxnet/ndarray/numpy/_op.py

```diff
@@ -7482,14 +7482,8 @@ def where(condition, x=None, y=None):  # pylint: disable=too-many-return-stateme
             else:
                 return y
         else:
-            if isinstance(x, numeric_types) and isinstance(y, numeric_types):
-                return _npi.where_scalar2(condition, float(x), float(y), out=None)
-            elif isinstance(x, NDArray) and isinstance(y, NDArray):
-                return _npi.where(condition, x, y, out=None)
-            elif isinstance(y, NDArray):
-                return _npi.where_lscalar(condition, y, float(x), out=None)
-            elif isinstance(x, NDArray):
-                return _npi.where_rscalar(condition, x, float(y), out=None)
+            if isinstance(x, NDArray) and isinstance(y, NDArray):
```

Comment:
Seems `_api_internal.where` handles scalar cases. Why we do we need to check `isinstance(x, NDArray)` here?

Reply:
There's really no need to check